docs/enhancements/1-crds-and-controllers.md

```diff
@@ -0,0 +1,430 @@
+      # If true, then a service will be exposed as <original-svc-name>.<original-svc-ns>.<search-domain>
+      includeNamespace: false
+      # Examples:
+      # 1. `searchDomain: svc.west.mesh` and `includeNamespace: true` will export ratings and reviews as ratings.bookinfo.svc.west.mesh and reviews.bookinfo.svc.west.mesh
+      # 2. `searchDomain: mesh.global` and `includeNamespace: false` will export ratings and reviews as ratings.mesh.global and reviews.mesh.global.
+  # Optional
+  # Empty import allows to not subscribe FDS.
+  # This rule imports only ratings service from any namespace in the west cluster.
+  import:
+  - meshID: west
+    # Optional
+    # If no selector is specified, then everything is imported.
+    serviceSelectors:
+    - matchExpressions:
+      - key: app.kubernetes.io/name
+        operator: In
+        values:
+        - ratings
+  # This rule imports everything from the central cluster.
+  - meshID: central
+```
+
+Rules based on label selectors are useful when an admin wants to give users the control on exporting Services,
+while `serviceList` gives the admin full control on exported services.
+
+We are able to provide the same functionality with only label selectors for services and namespaces.
+In this variant, `export` is a list of rule sets containing `serviceSelector` or `namespaceSelector`.
+Rules are OR-ed, but each rule set (an item in the `export` list) is AND-ed.
+
+```yaml
+apiGroup: federation.openshift-service-mesh.io/v1alpha1
+kind: FederationServiceRules
+metadata:
+  # Name must be default.
+  name: default
+  # Namespace is expected to be the same as the controller's namespace.
+  namespace: istio-system
+spec:
+  # The following list of rules exports services "ratings" and "reviews" from namespace "ns-1" and service "details" from namespace "ns-2".
+  export:
+    selectors:
+    - serviceSelector:
+        matchExpressions:
+        - key: app.kubernetes.io/name
+          operator: In
+          values:
+          - ratings
+          - reviews
+      namespaceSelector:
+        matchLabels:
+          kubernetes.io/metadata.name: ns-1
+    - serviceSelector:
+        matchLabels:
+          app.kubernetes.io/name: details
+      namespaceSelector:
+        matchLabels:
+          kubernetes.io/metadata.name: ns-2
+    dnsSettings:
+      searchDomain: mesh.global
+  # Optional
+  # Empty import allows to not subscribe FDS.
+  import:
+  - meshID: west
+    # Optional
+    # If no selector is specified, then everything is imported.
+    serviceSelectors:
+    - matchExpressions:
+      - key: app.kubernetes.io/name
+        operator: In
+        values:
+        - ratings
+```
+
+Responsibilities of the controller for this resource:
+1. Owning resources for exported services:
+   * `Gateway`;
+   * `ServiceEntry` - if custom search domain configured;
+   * `EnvoyFilter` - if the ingress type is "openshift-router".
+2. Watching services and namespaces and check if they match export rules.
```

Comment:
I think we should rephrase it - it's not about checking if they match our rules, but we are watching svc and ns that match defined rules and coordinates exporting between the meshes.